libsrc/webconfig/StaticFileServing.cpp

```diff
@@ -115,24 +119,44 @@ void StaticFileServing::onRequestNeedsReply (QtHttpRequest * request, QtHttpRepl
 		QStringList uri_parts = path.split('/', QString::SkipEmptyParts);
 
 		// special uri handling for server commands
-		if ( ! uri_parts.empty() && uri_parts.at(0) == "cgi"  )
+		if ( ! uri_parts.empty() )
 		{
-			uri_parts.removeAt(0);
-			try
+			if(uri_parts.at(0) == "cgi")
 			{
-				_cgi.exec(uri_parts, request, reply);
+				uri_parts.removeAt(0);
+				try
+				{
+					_cgi.exec(uri_parts, request, reply);
+				}
+				catch(int err)
+				{
+					Error(_log,"Exception while executing cgi %s :  %d", path.toStdString().c_str(), err);
+					printErrorToReply (reply, QtHttpReply::InternalError, "script failed (" % path % ")");
+				}
+				catch(std::exception &e)
+				{
+					Error(_log,"Exception while executing cgi %s :  %s", path.toStdString().c_str(), e.what());
+					printErrorToReply (reply, QtHttpReply::InternalError, "script failed (" % path % ")");
+				}
+				return;
 			}
-			catch(int err)
+			else if ( uri_parts.at(0) == "json-rpc" )
 			{
-				Error(_log,"Exception while executing cgi %s :  %d", path.toStdString().c_str(), err);
-				printErrorToReply (reply, QtHttpReply::InternalError, "script failed (" % path % ")");
+				QMetaObject::Connection m_connection;
+				QByteArray data = request->getRawData();
+				QtHttpRequest::ClientInfo info = request->getClientInfo();
+
+				m_connection = QObject::connect(_jsonProcessor, &JsonProcessor::callbackMessage,
```

Comment:
@Paulchen-Panther 
die QT5 + C++11 syntax hat mich gerettet, nen slot kann man in qt4 so inline garnicht abfangen und dazu noch ne lokale variable (reply) reindrücken oder?